isis/src/base/apps/isis2pds/isis2pds.cpp

```diff
@@ -0,0 +1,244 @@
+#include <QString>
+
+#include "Cube.h"
+#include "ExportDescription.h"
+#include "FileName.h"
+#include "Process.h"
+#include "ProcessExportPds.h"
+#include "ProcessExportPds4.h"
+#include "Pvl.h"
+#include "PvlKeyword.h"
+#include "PvlToXmlTranslationManager.h"
+#include "isis2pds.h"
+
+using namespace std;
+namespace Isis{
+
+  enum Pixtype { NONE, NEG, BOTH };
+
+  void setRangeAndPixels(UserInterface &ui, ProcessExport &p,
+                         double &min, double &max, Pixtype ptype);
+  void isis2pds(UserInterface &ui, Pvl *log) {
```

Comment:
We should add a second isis2pds function with a cube parameter. Follow the steps on the [testing page](https://github.com/USGS-Astrogeology/ISIS3/wiki/Writing-ISIS3-Tests-Using-Gtest-and-Ctest#creating-a-basic-callable-function) or check out the [grid app](https://github.com/USGS-Astrogeology/ISIS3/blob/1fc23d07df1539d73a3aa118d748bf1d3f6ec67c/isis/src/base/apps/grid/grid.cpp#L32) for an example.

Reply:
Push updated code with added function. The only potential issue that I came across when comparing my test suite with the grid tests is that when I try calling isis2pds(testCube, options), it will interchangeably say segfault or child aborted. Tests pass if I keep the call as isis2pds(options). In order to prevent the segfault within isis2pds.cpp, I have to revert the code p.SetInputCube(icube) back to p.SetInputCube(ui.GetFileName("FROM"), inputAtt).

Reply:
I think this is because the cube isn't "open" (function (here)[https://github.com/USGS-Astrogeology/ISIS3/blob/87c8432b2c21f60610204b41a378f5c24464b6ca/isis/src/base/objs/Cube/Cube.cpp#L636]) before trying to set it as the input. To be able to read from a cube, it firsts needs to be opened with the `open` function. The `p.SetInputCube(ui.GetFileName("FROM"), inputAtt)` call opens the cube for you.

Reply:
When I try opening the cube before setting as input, I get all failed tests stating the the cube is already open.

Reply:
Discussed this issue in standup. I will look into this more today regarding what is causing the segfault. Since I will not be working the rest of the week, we decided that if it is not resolved by the end of today, to just revert it back to not having a function with the cube parameter.

Reply:
Resolved segfault by switching tests from SmallCube to DefaultCube.